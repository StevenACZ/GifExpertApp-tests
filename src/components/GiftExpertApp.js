import React, { useState } from 'react';
import AddCategory from './AddCategory';
import GifGrid from './GifGrid';

const GiftExpertApp = () => {
  const [categories, setCategory] = useState(['Goku']);

  return (
    <div>
      <h1>GiftExpertApp</h1>
      <AddCategory setCategory={ setCategory } />      
      <hr />

      { 
        categories.map( category => (
          <GifGrid
            key={ category }
            category={ category }
          />
        ))
      }
    </div>
  )
};

export default GiftExpertApp;