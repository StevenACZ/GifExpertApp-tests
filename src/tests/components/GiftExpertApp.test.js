import react from 'react';
import { shallow } from "enzyme";
import GiftExpertApp from "../../components/GiftExpertApp";

describe('Prueba en <GiftExpertApp />', () => {
  test('debe de mostrar el elemento', () => {
    const wrapper = shallow( <GiftExpertApp /> );

    expect( wrapper ).toMatchSnapshot();
  });

  test('debe de mostrar una lista de categorias', () => {
    const categories = ['One punch', 'Dragon ball'];
    const wrapper = shallow( <GiftExpertApp defaultCategories={ categories } /> );
    
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length ).toBe( categories.length );
  });
});