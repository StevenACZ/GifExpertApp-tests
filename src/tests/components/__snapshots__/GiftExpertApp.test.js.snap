// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GiftExpertApp /> debe de mostrar el elemento 1`] = `
<div>
  <h1>
    GiftExpertApp
  </h1>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
</div>
`;

exports[`Prueba en <GiftExpertApp /> debe de mostrar una lista de categorias 1`] = `
<div>
  <h1>
    GiftExpertApp
  </h1>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <GifGrid
    category="One punch"
    key="One punch"
  />
  <GifGrid
    category="Dragon ball"
    key="Dragon ball"
  />
</div>
`;
